### PR TITLE
Rewrite testing examples of images in python

### LINKED ISSUE #20254: CI: Rewrite testing examples of images in python
We are runing automated tests for example images building as Bash scripts - those should be converted to Python.

https://github.com/apache/airflow/blob/98514cc1599751d7611b3180c60887da0a25ff5e/.github/workflows/ci.yml#L322

### PR DESCRIPTION
**Pull request by edithturn**
Rewrite testing examples of images in python

01. **Change the script to python and eliminate unnecessary scripts:**
     - airflow/docker_tests/test-examples-of-prod-image-building.sh to  airflow/docker_tests/test-examples-of-prod-image-building.py
     - Removed the call to scripts/ci/images/ci_run_docker_tests.sh
02. **Added working-directory to build other dependencies of test examples prod:**
     - The script is running in his directory: `DOCKER_EXAMPLES_DIR = SOURCE_ROOT / "docs" / "docker-stack" / "docker-examples"`, so it is necesary specify the working-directory in ci.yaml in the same directory to complete the call of more dependencies.
     - Added "requests" as a requirement in setup.cfg, this is importing in the final script:` import requests`

Related to: 
- PR: https://github.com/apache/airflow/pull/20913
- TASK: https://github.com/apache/airflow/issues/20254

Close: #20254

### CODE REVIEW

**Comment by edithturn** [comment]
@potiuk I tested, my mistake is that I was calling to this path in the working-directory:  **tests** since there is where **conftest.py** file is. But the code of test_examples_of_prod_image_building.py says exactly where the tests are running.

I also compare (left are my PR and right is how it was working before)
* It is executing all the tests (passed and skipped are the same quantity). Except for **Warnings** are not the same, I don't think they need to give us the same amount always.
* Another difference is "creating a virtual environment" which is part of the code ci_run_docker_tests.sh, which we don't need it

**Comment by edithturn** [comment]
@potiuk this is already rebase and squashed.

File: dev/breeze/setup.cfg [hunk 1]
```diff
@@ -58,6 +58,7 @@ install_requires =
     pytest-xdist
     rich
     click_completion
+    requests
```
Comment (potiuk) [inline dev/breeze/setup.cfg:61]:
```suggestion
    requests
```

I think it can be removed :).

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
LGTM :). Just one small thing with "requests" whch (I believe) are not needed?

**Comment by edithturn** [comment]
@potiuk without "request" in the setup.cfg, we will have this error, I think it is necesary.

Screenshot from 2022-01-26 15-16-29

File: dev/breeze/setup.cfg (hunk 1 above)
Comment (potiuk) [reply dev/breeze/setup.cfg:61]: > @potiuk without "request" in the setup.cfg, we will have this error, I think it is necesary.

Indeed :). Merging

**Comment by edithturn** [comment]
Thank you @potiuk 🙏🏾💪🏾🎉

**Pull request merged**